lib/internal/http2/core.js

```diff
@@ -2897,7 +2897,7 @@ function connect(authority, options, listener) {
   } else {
     switch (protocol) {
       case 'http:':
-        socket = net.connect(options.port || port, options.host || host);
+        socket = net.connect({ port, host, ...options });
```

Comment:
Is it intentional that `options.host` and `options.port` would now take precedence over those same fields in `authority`?

Reply:
> Is it intentional that options.host and options.port would now take precedence over those same fields in authority

Yes, this behavior was already changed in https://github.com/nodejs/node/pull/28584